### PR TITLE
MINIFICPP-1359 Implement FollowRedirects property in InvokeHTTP

### PR DESCRIPTION
**Pull request by lordgamez**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/http-curl/tests/VerifyInvokeHTTPTest.cpp [hunk 1]
```diff
@@ -120,6 +126,31 @@ class VerifyInvokeHTTPOKResponse : public VerifyInvokeHTTP {
   }
 };
 
+class VerifyInvokeHTTPOK200Response : public VerifyInvokeHTTP {
+ public:
+  void runAssertions() override {
+    using org::apache::nifi::minifi::utils::verifyLogLinePresenceInPollTime;
+    assert(verifyLogLinePresenceInPollTime(std::chrono::seconds(6),
+        "key:invokehttp.status.code value:200",
+        "response code 200"));
```
Comment (adamdebreceni) [inline extensions/http-curl/tests/VerifyInvokeHTTPTest.cpp:135]: which handler responds with `200`? we redirect to `/`, does the civetweb have some default handler on `/`?

Comment (lordgamez) [reply extensions/http-curl/tests/VerifyInvokeHTTPTest.cpp:135]: It seems so. At first I wanted to make a separate handler for it, but after seeing it automatically returns `HTTP 200 OK` after redirecting to `/` and no other handler returns code `200` I decided to use that.

**Review by arpadboda — APPROVED** [review]

File: extensions/http-curl/processors/InvokeHTTP.cpp [hunk 2]
```diff
@@ -75,7 +75,11 @@ core::Property InvokeHTTP::ReadTimeout(
 core::Property InvokeHTTP::DateHeader(
     core::PropertyBuilder::createProperty("Include Date Header")->withDescription("Include an RFC-2616 Date header in the request.")->isRequired(false)->withDefaultValue<bool>(true)->build());
 
-core::Property InvokeHTTP::FollowRedirects("Follow Redirects", "Follow HTTP redirects issued by remote server.", "True");
+core::Property InvokeHTTP::FollowRedirects(
+  core::PropertyBuilder::createProperty("Follow Redirects")
+  ->withDescription("Follow HTTP redirects issued by remote server.")
+  ->withDefaultValue<bool>(true)
```
Comment (arpadboda) [inline extensions/http-curl/processors/InvokeHTTP.cpp:81]: According to Curl, this is false by default:
https://curl.se/libcurl/c/CURLOPT_FOLLOWLOCATION.html

So I think this is a breaking change, I would prefer to have false here by default.

Comment (lordgamez) [reply extensions/http-curl/processors/InvokeHTTP.cpp:81]: That's true, but I was checking the usage of this property in NiFi's version of [InvokeHTTP](https://nifi.apache.org/docs/nifi-docs/components/org.apache.nifi/nifi-standard-nar/1.13.2/org.apache.nifi.processors.standard.InvokeHTTP/index.html) and there the default value is True. Should we stil prefer the Curl's version for the default value?

Comment (arpadboda) [reply extensions/http-curl/processors/InvokeHTTP.cpp:81]: Hmm, good question. 
My reasoning was not based on Curl's default, but what we did before (which was based on Curl's default).
As minifi is still 0.xxx version, I tend to agree with you and introduce the breaking change to align with NiFi in this aspect.

**Review by adamdebreceni — APPROVED** [review]

**Review by arpadboda — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Pull request closed without merging**